### PR TITLE
added 3 new testes and updated doc for 4th test

### PR DESCRIPTION
**Pull request by raya311**
# Pull Request

## Summary

Added 3 new tests for git issues 98,121 and 122

## Why

To increase the regression coverage. For more details, please refer to week13-tests.md

## How to Verify

Run the tests using command 'npx playwright test'

1. Install dependencies: `npm install`
2. Run the app / relevant command
3. Run lint: `npm run eslint`
4. Run tests: `npm run test`
5. Describe expected behavior

## Checklist (Definition of Done)

- Branch follows team naming pattern (`week<NUM>/<short-description>`)
- PR opened early and kept reviewable in size
- At least one teammate review requested
- CI checks are passing
- Lint passes locally (`npm run eslint`)
- Tests pass locally (`npm run test`)
- New behavior includes starter tests or test plan notes
- Documentation updated if needed (`README`, `docs/`, ADRs)

## Notes for Reviewers

Anything specific you'd like feedback on?
- Edge cases?
- Architecture concerns?
- Naming?
- Test coverage?

### CODE REVIEW

File: docs/final/week13-tests.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+## Overview
+
+Four critical regression tests have been added to strengthen trust in the system by protecting against regressions in recently fixed bugs and weak spots. These tests ensure that authentication error handling, authError redirect behavior, form change detection, and edit-mode state management remain reliable.
+
+---
+
+## Test Inventory: Week 13 Additions
+
+### Total New Tests Added: 4
```
Comment (hmhawn) [inline docs/final/week13-tests.md]: change “4 new tests” to “3 new tests and 1 improved existing regression test”

File: docs/final/week13-tests.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+8. Verify button becomes enabled after edit
+
+**Recent Fix It Protects:**
+- **Issue:** Edit-mode UI state was inconsistent when loading saved trips (flaky behavior)
+- **Fix Applied:** Improved state management test to verify disabled/enabled state transitions
+- **Impact:** Ensures users can reliably edit trips with consistent button states; prevents silent regressions in edit-mode behavior
+
+---
+
+### 4. Issue #122: Change Detection Tests - Form State Tracking
```
Comment (hmhawn) [inline docs/final/week13-tests.md:115]: I think the Issue #122 test needs to be strengthened because it overlaps quite a bit with the #126 test but doesn’t currently add much additional coverage.

The #126 test already verifies the key edit-mode behavior end-to-end, including that the Update button is disabled after loading a trip and only becomes enabled after a real edit. That already covers a significant portion of change-detection behavior.

The Issue #122 test, as written, mostly verifies that a trip can be edited and updated successfully, which is already covered by other workflow tests and partially by #126. It doesn’t clearly isolate or prove the actual change-detection logic.

Since Issue #122 is specifically about change detection, I think the test should more explicitly validate that behavior. For example:

* verify the Update button is disabled immediately after loading a trip
* verify it becomes enabled after a field is changed
* ideally, verify it becomes disabled again if the field is reverted to its original value

That would make the test clearly focused on state comparison rather than just update success, and ensure it provides distinct regression protection instead of overlapping with #126.

**Review by hmhawn — CHANGES_REQUESTED** [review]
Please see comments

File: docs/final/week13-tests.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+Four critical regression tests have been added to strengthen trust in the system by protecting against regressions in recently fixed bugs and weak spots. These tests ensure that authentication error handling, authError redirect behavior, form change detection, and edit-mode state management remain reliable.
+
+---
+
+## Test Inventory: Week 13 Additions
+
+### Total New Tests Added: 3 and 1 improved existing regression test
+- **Issue #98:** [tests/e2e/auth-error.spec.js](tests/e2e/auth-error.spec.js#L5) - Auth failure error feedback
+- **Issue #121:** [tests/e2e/auth-error.spec.js](tests/e2e/auth-error.spec.js#L33) - Auth error access control
+- **Issue #126 (PR #130):** [tests/e2e/primary-workflow.spec.js](tests/e2e/primary-workflow.spec.js#L219) - Edit-mode state management - impreoved test
```
Comment (hmhawn) [inline docs/final/week13-tests.md]: typo here impreoved needs corrected to improved

**Review by hmhawn — APPROVED** [review]
approved but small typo should be corrected (see note)
